### ISSUE TITLE
D10 readiness for site

### ISSUE THREAD

**Issue by thinkcollege** [communityinclusion/escoach#105]
I made some changes on the devel branch to get things ready for D10

1. Uninstalled simple_mail module
2. upgraded nodeaccess and menu_breadcrumb to alpha versions: working fine
3. slight change to info file for field_hidden module to make it D10 compliant
The only blockers for D10 migration are serialization module, which has to be uninstalled and upgraded after migration, and the admin_can_login_anyuser module. The latter needs to go away.  It has security issues and will be replaced by @ultimike 's custom modifications to allow admin users to assign providers to TA users.  There is also @ultimike 's ES_Migrate custom module.  I don't know if you need that any more.  You can remove it from devel or make it ready for D10

**Comment by ultimike** [comment]
Feel free to uninstall the `es_migrate` custom module, it is no longer necessary. 

We previously created the `es_home` branch by branching off of `devel` - unless you say otherwise, we will rebase `devel` onto `es_home`. Also see #106 

-mike

**Comment by thinkcollege** [comment]
@ultimike @mgcobblestone  I have the devel branch almost ready to go for D10.  The CSV Serialization module has to be installed after migration.  The ES Homepage module is showing a small error on the upgrade status report.  I would like to get the development branches migrated and leave the live site (rack) branch on D9 because the owners need that admin can login anyuser module--just until we are ready to put the new site in place.  Can we coordinate on the migration to D10 on devel and es_home?

**Comment by ultimike** [comment]
@thinkcollege I just updated the `es_home` module to be D10 compatible on the `es_home` branch - it was just some (very) minor stuff.

Anything else I can do to assist?

-mike

**Comment by thinkcollege** [comment]
I am ready to migrate the devel branch to D10.  I don't want to run into conflicts with the es_home branch.  Do you have any helpful hints on Composer stuff for this site?  Generally I find the 9 to 10 migration to be a snake's nest of Composer glitches.

**Comment by ultimike** [comment]
Yeah, so what I normally do is this:

1. First make sure all enabled modules and themes are D10 ready according to Upgrade Status. 
2. If you have modules that need a major upgrade to be D10 ready, then do a `composer remove` on those first. 
3. Then, I normally `composer remove` drush and any other dependencies that could be problematic. 
4. Manually update composer.json file with 10s for drupal/core-* stuff. 
5. Run `composer update drupal/core-* drupal/upgrade_status --with-all-dependencies`
6. Run `composer require drush/drush` and on any other dependencies you removed in step 2 above.

Your milage may vary. 

In short, I found that temporarily using `composer remove` on potentially problematic dependencies goes a long way.

-mike

**Comment by thinkcollege** [comment]
@ultimike devel branch is running Drupal 10, deployed to rp-escoach.communityinclusion.org. Do you want to get es_home up to D10 so we won't have any problem merging changes into devel?

**Comment by ultimike** [comment]
@thinkcollege I am working on rebasing `devel` into `es_home` - I believe I have it almost working, but after the rebase, I'm getting an error about missing modules: `admin_can_login_anyuser` and `phpass` -are these Drupal 9 or Drupal 10 modules?

thanks,
-mike

**Comment by thinkcollege** [comment]
admin_can_login_anyuser is a D9 compatible module that did the work of allowing admins to switch users so they could look at other users' dashboard.  We are getting rid of that to be replaced by your dashboard work.  It is also not covered by Drupal security policy.  So, I uninstalled and removed it.  I don't know anything about phpass: https://www.drupal.org/project/phpass  I don't think I ever installed it and I haven't come across it during the migration.

**Comment by ultimike** [comment]
@thinkcollege Thanks - that was helpful.

I believe that I have rebased the `devel` branch into `es_home`, so that you should be able to do an easy (fast-forward) merge of `es_home` into `devel`. 

For now, please don't do anything, as @mgcobblestone is going to test the rebase in the morning. If he agrees that I didn't break anything, then you'll be all clear to do the merge. 

Sound good?

-mike

**Comment by thinkcollege** [comment]
A-okay

**Comment by ultimike** [comment]
@thinkcollege Martin reports no issues with the `es_home` branch, feel free to merge it into `devel` at your convenience. 

-mike

**Comment by ultimike** [comment]
@thinkcollege Can this issue be closed?

-mike
